### PR TITLE
Add commands for Network Security Perimeter

### PR DESCRIPTION
**Pull request by nikhilsaxena2609**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Network Security Perimeter

**Comment by nikhilsaxena2609** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

File: src/nsp/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Azure CLI Nsp Extension #
+This is an extension to Azure CLI to manage Nsp resources.
```
Comment (necusjz) [inline src/nsp/README.md:2]:
```suggestion
# Azure CLI NSP Extension #
This is an extension to Azure CLI to manage NSP resources.
```

File: src/nsp/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+## How to use ##
```
Comment (necusjz) [inline src/nsp/README.md:4]: Please complete this section.

Comment (nikhilsaxena2609) [reply src/nsp/README.md:4]: Hi @necusjz , based on the command I used locally, "azdev extension add nsp", I've added "az extension add --name nsp". Can you please confirm where the name 'nsp' will come from and if this is correct?

Comment (necusjz) [reply src/nsp/README.md:4]: It may come from your custom name of the folder. As it stands for "Network Security Perimeter", make sense.

**Pull request merged**